### DISCUSSION TITLE
Error while installing Neovim 0.10 with Homebrew on Ubuntu : Need Help

### DISCUSSION THREAD

**Discussion by Trid-collab** [Homebrew/discussions#4491] [Linux]
### Additional Info

- [X] I am using Windows Subsystem for Linux

### Output of `brew config`

```shell
brew config
HOMEBREW_VERSION: 4.0.17
ORIGIN: https://github.com/Homebrew/brew
HEAD: c0f8068573bbf886fdc78d3f9daca54d53e7e49e
Last commit: 3 hours ago
Core tap JSON: 08 May 10:54 UTC
HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: :0
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.10 => /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.10_1/bin/ruby
CPU: octa-core 64-bit unknown_0x6_0x8c
Clang: N/A
Git: 2.40.1 => /home/linuxbrew/.linuxbrew/bin/git
Curl: 7.81.0 => /bin/curl
Kernel: Linux 5.15.90.1-microsoft-standard-WSL2 x86_64 GNU/Linux
OS: Ubuntu 22.04.2 LTS (jammy)
WSL: 2 (Microsoft Store)
Host glibc: 2.35
/usr/bin/gcc: 11.3.0
/usr/bin/ruby: N/A
glibc: N/A
gcc@11: N/A
gcc: N/A
xorg: N/A
```

### Output of `brew doctor`

```shell
brew doctor
Your system is ready to brew.
```

### Description of issue

For the past few weeks I have facing the following errors when I am trying to install neovim nightly 0.10 on Ubuntu using HOMEBREW

brew install neovim --head

I don't face errors when I use the stable version of neovim 0.9

brew install neovim

The errors I have received is as follows: Need help

`Fetching neovim
==> Downloading https://raw.githubusercontent.com/Homebrew/homebrew-core/be2dda92f7e3a105670078019f22e8c0a1c379c9/Formula/neovim.rb
Already downloaded: /home/tridib/.cache/Homebrew/downloads/a0766852b8e220744893d542af12ef3cdf9f84b21aedf83760c9ca61b694f962--neovim.rb
==> Downloading https://luarocks.org/manifests/gvvaughan/lpeg-1.0.2-1.src.rock
Already downloaded: /home/tridib/.cache/Homebrew/downloads/0bc17975f0cf3981083172aa7da329b874e5780fa81494f5dd07d0665fcec23b--lpeg-1.0.2-1.src.rock
==> Cloning https://github.com/neovim/neovim.git
Updating /home/tridib/.cache/Homebrew/neovim--git
==> Checking out branch master
Already on 'master'
Your branch is up to date with 'origin/master'.
HEAD is now at 4ecf6fdfd fix(statusline): bail out properly on negative row (#23535)
==> cmake -S . -B build -DCMAKE_INSTALL_PREFIX=/tmp/neovim-20230508-27753-4pzpyf/deps-build
==> cmake --build build
==> cmake --install build
==> cmake -S . -B build -DLIBLUV_LIBRARY=/home/linuxbrew/.linuxbrew/opt/luv/lib/libluv.so -DLIBUV_LIBRARY=/home/linuxbrew/.linuxbrew/opt/libuv/lib/libuv.so
==> cmake --build build
Last 15 lines from /home/tridib/.cache/Homebrew/Logs/neovim/05.cmake:
-- Using NVIM_VERSION: v0.10.0-dev-1656+g4ecf6fdfd-Homebrew
[  3%] Linking C shared module ../../lib/libnlua0.so
gmake[2]: Leaving directory '/tmp/neovim-20230508-27753-4pzpyf/build'
cd /tmp/neovim-20230508-27753-4pzpyf/build/src/nvim && /home/linuxbrew/.linuxbrew/Cellar/cmake/3.26.3/bin/cmake -E cmake_link_script CMakeFiles/nlua0.dir/link.txt --verbose=1
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/shims/linux/super/gcc-11 -fPIC -I/home/linuxbrew/.linuxbrew/opt/luajit/include/luajit-2.1 -O3 -DNDEBUG -shared  -o ../../lib/libnlua0.so CMakeFiles/nlua0.dir/__/mpack/conv.c.o CMakeFiles/nlua0.dir/__/mpack/lmpack.c.o CMakeFiles/nlua0.dir/__/mpack/mpack_core.c.o CMakeFiles/nlua0.dir/__/mpack/object.c.o CMakeFiles/nlua0.dir/__/mpack/rpc.c.o CMakeFiles/nlua0.dir/__/nlua0.c.o  /tmp/neovim-20230508-27753-4pzpyf/deps-build/lib/liblpeg.a
[  3%] Built target update_version_stamp
/home/linuxbrew/.linuxbrew/opt/binutils/bin/ld: /tmp/neovim-20230508-27753-4pzpyf/deps-build/lib/liblpeg.a(lpcode.c.o): warning: relocation against `numsiblings' in read-only section `.text'
/home/linuxbrew/.linuxbrew/opt/binutils/bin/ld: /tmp/neovim-20230508-27753-4pzpyf/deps-build/lib/liblpeg.a(lptree.c.o): relocation R_X86_64_PC32 against symbol `numsiblings' can not be used when making a shared object; recompile with -fPIC
/home/linuxbrew/.linuxbrew/opt/binutils/bin/ld: final link failed: bad value
collect2: error: ld returned 1 exit status
gmake[2]: *** [src/nvim/CMakeFiles/nlua0.dir/build.make:181: lib/libnlua0.so] Error 1
gmake[2]: Leaving directory '/tmp/neovim-20230508-27753-4pzpyf/build'
gmake[1]: *** [CMakeFiles/Makefile2:565: src/nvim/CMakeFiles/nlua0.dir/all] Error 2
gmake[1]: Leaving directory '/tmp/neovim-20230508-27753-4pzpyf/build'
gmake: *** [Makefile:159: all] Error 2`

**Comment by guysherman** [comment]
You might be able to go `CFLAGS=-fPIC brew install --head neovim`  I don't know if it will pass that environment variable all the way down to the gcc processes, but worth a shot.

**Comment by osalbahr** [comment]
My guess is that this has to do with your system’s gcc version. What does `/usr/bin/gcc --version` tell you?

**Reply by osalbahr** [reply]
Does running [`sudo apt-get install build-essential procps curl file git`](https://docs.brew.sh/Homebrew-on-Linux#requirements) change the error message?

(This probably won’t help, but doesn’t hurt to double check)

**Reply by Trid-collab** [reply]
Right now after couple of update it seems that the issue has been resolved. Thank you
